lib/Jit/LLILCJit.cpp

```diff
@@ -102,7 +109,9 @@ LLILCJit::LLILCJit() {
   InitializeNativeTargetAsmParser();
   llvm::linkStatepointExampleGC();
 
-  ShouldUseConservativeGC = shouldUseConservativeGC();
+  ShouldInsertStatepoints = shouldInsertStatepoints();
+  assert(ShouldInsertStatepoints || shouldUseConservativeGC() &&
+    "Statepoints not inserted in precise GC mode");
```

Comment:
I would change the wording here to "Statepoints are required for precise GC" or similar. The current message leaves me with the impression that statepoints should not be inserted for precise GC.